src/planner/binder/query_node/plan_setop.cpp

```diff
@@ -23,14 +24,39 @@ unique_ptr<LogicalOperator> Binder::CastLogicalOperatorToTypes(vector<LogicalTyp
 	if (node->type == LogicalOperatorType::LOGICAL_PROJECTION) {
 		// "node" is a projection; we can just do the casts in there
 		D_ASSERT(node->expressions.size() == source_types.size());
+		if (node->children.size() == 1 && node->children[0]->type == LogicalOperatorType::LOGICAL_GET) {
+			// Is this a CSV Scan?
+			auto &logical_get = node->children[0]->Cast<LogicalGet>();
+			// yuck string matching, maybe there is a better way?
+			if (logical_get.function.name == "read_csv" || logical_get.function.name == "read_csv_auto") {
+				// We loop in the expressions and bail out if there is anything weird (i.e., not a bound column ref)
+				bool can_pushdown_types = true;
+				for (auto &expr : op->expressions) {
+					can_pushdown_types = can_pushdown_types && expr->type == ExpressionType::BOUND_COLUMN_REF;
```

Comment:
We should look at the `BoundColumnRefExpression` here to do the mapping instead of assuming `logical_get.column_ids` are in the same order (although that might even always be the case here for these straightforward queries)